src/Microsoft.TemplateEngine.Edge/Settings/LocalizationCacheKeyComparer.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.TemplateEngine.Edge.Settings
+{
+    /// <summary>
+    /// Makes an ordinal, case insensitive comparison with the two strings of the given tuples.
+    /// </summary>
+    internal class LocalizationCacheKeyComparer : IEqualityComparer<(string? Key, string? Value)>
```

Comment:
I agree. I only do it this way because it is "the correct way". I would like to understand why this is considered a good practice and whether we can ignore it in cases such as this one.